library/ssl_tls.c

```diff
@@ -2872,6 +2866,8 @@ int ssl_parse_certificate( ssl_context *ssl )
             if( pk_can_do( pk, POLARSSL_PK_ECKEY ) &&
                 ! ssl_curve_is_acceptable( ssl, pk_ec( *pk )->grp.id ) )
             {
+                ssl->session_negotiate->verify_result |= BADCERT_BAD_EC;
```

Comment:
@andresag01 After our discussion, I have renamed `BADCERT_BAD_EC` to a generic `BADCERT_BAD_KEY` using the same error code as the later mbed TLS versions.